### PR TITLE
Closes #887: Use LeakCanary in Dev (And Beta?) builds

### LINKED ISSUE #887: Use LeakCanary in Dev (And Beta?) builds
In #877 we saw multiple instances of MainActivity. LeakCanary can help us to see whether it's our code that keeps references to old instances.

We should enable it in Dev builds. In beta builds it would be helpful too - but it might be confusing to our testers.

### PR DESCRIPTION
**Pull request by siddware**
Integrated Leakcanary for debug builds. Closes #887

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/FocusApplication.java [hunk 1]
```diff
@@ -6,9 +6,12 @@
 package org.mozilla.focus;
 
 import android.arch.lifecycle.LiveData;
+import android.content.Context;
```
Comment (sputnikci) [inline app/src/main/java/org/mozilla/focus/FocusApplication.java]: \[Checkstyle\] ERROR: Unused import \- android\.content\.Context\.

Comment (siddware) [reply app/src/main/java/org/mozilla/focus/FocusApplication.java]: Thanks, resolved it :)

File: app/src/main/java/org/mozilla/focus/FocusApplication.java [hunk 2]
```diff
@@ -25,9 +27,12 @@
 
 public class FocusApplication extends LocaleAwareApplication {
     private VisibilityLifeCycleCallback visibilityLifeCycleCallback;
-
```
Comment (mcomella) [inline app/src/main/java/org/mozilla/focus/FocusApplication.java]: nit: please keep the whitespace

File: build.gradle [hunk 3]
```diff
@@ -56,6 +57,16 @@ task clean(type: Delete) {
 configurations {
     ktlint
 }
+// To ensure unit tests and instrumentation tests remain consistent after the introduction of leakcanary
```
Comment (mcomella) [inline build.gradle]: nit: whitespace above this line

Comment (mcomella) [inline build.gradle]: nit: I agree with the sentiment – let's not change the tests just yet. Can you:
- file an issue (you don't have to work on it) to investigate whether we want to add this to unit or UI tests
- Mention "See #xxx" in this comment where xxx is the issue I mention above.

Comment (siddware) [reply build.gradle]: See #2041

**Comment by mcomella** [comment]
I can confirm this appears when using a debug build on a device (but not on the emulator, for some reason).

**Pull request merged**